oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzureUtilities.java

```diff
@@ -159,6 +134,38 @@ public static CloudBlobDirectory cloudBlobDirectoryFrom(String connection, Strin
         return container.getDirectoryReference(dir);
     }
 
+    private static ResultSegment<ListBlobItem> listBlobsInSegments(CloudBlobDirectory directory,
+           ResultContinuation token) throws IOException {
+        ResultSegment<ListBlobItem> result = null;
+        IOException lastException = null;
+        for (int sleep = 10; sleep <= 10000; sleep *= 10) {  //increment the sleep time in steps.
+            try {
+                result = directory.listBlobsSegmented(
+                        null,
+                        false,
+                        EnumSet.of(BlobListingDetails.METADATA),
+                        5000,
+                        token,
+                        null,
+                        null);
+                break;  //we have the results, no need to retry
+            } catch (StorageException | URISyntaxException e) {
+                lastException = new IOException(e);
+                try {
+                    Thread.sleep(sleep); //Sleep and retry
+                } catch (InterruptedException ex) {
+                    log.warn("Interrupted", e);
+                }
+                continue;
```

Comment:
I don't think "continue" is needed her

Reply:
+1. I think, I missed it in refactoring.